acra/configuring-maintaining/tls/cert_gen_with_openssl.md

```diff
@@ -0,0 +1,163 @@
+[ req ]
+distinguished_name = req_dn
+
+[ req_dn ]
+
+# X.509 v3 extensions for issued certificates
+[ v3_req ]
+basicConstraints = CA:false
+keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+subjectAltName = DNS:localhost
+
+# X.509 v3 extensions for CA certificate
+[ v3_ca ]
+basicConstraints = critical,CA:true
+subjectAltName = DNS:localhost
+subjectKeyIdentifier=hash
+authorityKeyIdentifier=keyid:always,issuer
+
+[ ca ]
+default_ca = CA_default
+
+[ CA_default ]
+dir = .
+database = $dir/index.txt
+serial = $dir/serial
+
+certificate = $dir/ca.crt.pem
+private_key = $dir/ca.key.pem
+
+default_md = sha256
+default_crl_days = 30
+policy = policy_anything
+
+[ policy_anything ]
+countryName             = optional
+stateOrProvinceName     = optional
+localityName            = optional
+organizationName        = optional
+organizationalUnitName  = optional
+commonName              = supplied
+emailAddress            = optional
+```
+
+## Create self-signed CA certificate
+
+```
+openssl req \
+    -x509 -sha256 -nodes -days 365 -newkey rsa:4096 \
+    -keyout ca.key.pem \
+    -out ca.crt.pem \
+    -subj '/C=GB/ST=London/L=London/O=Global Security/OU=IT/CN=Test CA'
+
+# Create certificate database, will be needed when signing CSRs
+touch index.txt
+```
+
+## Show generated certificate in readable format
+
+```
+openssl x509 -noout -text -in ca.crt.pem
+```
+Example output:
+```
+Certificate:
+    Data:
+        Version: 3 (0x2)
+        Serial Number:
+            64:28:80:32:72:4e:a6:5a:cc:31:bb:00:e5:bc:31:48:62:b1:e9:1f
+        Signature Algorithm: sha256WithRSAEncryption
+        Issuer: C = GB, ST = London, L = London, O = Global Security, OU = IT, CN = Test CA
+        Validity
+            Not Before: Jul  5 07:37:16 2021 GMT
+            Not After : Jul  5 07:37:16 2022 GMT
+        Subject: C = GB, ST = London, L = London, O = Global Security, OU = IT, CN = Test CA
+        Subject Public Key Info:
+            Public Key Algorithm: rsaEncryption
+                RSA Public-Key: (4096 bit)
+                Modulus:
+                    00:98:86:b9:17:5e:d2:80:64:68:83:4a:51:fd:c1:
+                    ### some lines skipped ###
+                    38:73:71:ad:64:5f:76:74:17:41:0d:5e:bf:e3:5a:
+                    a6:64:b1
+                Exponent: 65537 (0x10001)
+        X509v3 extensions:
+            X509v3 Subject Key Identifier:
+                80:17:42:A5:C4:C5:B5:5E:14:82:7D:72:74:25:0A:AB:C8:9F:E3:2E
+            X509v3 Authority Key Identifier:
+                keyid:80:17:42:A5:C4:C5:B5:5E:14:82:7D:72:74:25:0A:AB:C8:9F:E3:2E
+
+            X509v3 Basic Constraints: critical
+                CA:TRUE
+    Signature Algorithm: sha256WithRSAEncryption
+         8e:77:bf:da:4b:74:8c:c8:23:a2:a8:5b:9c:88:1b:58:90:96:
+         ### some lines skipped ###
+         f4:71:d2:7d:2c:6e:ae:01:5b:8c:2f:f8:70:a5:c5:b6:55:2e:
+         e9:00:62:5f:89:c3:0d:00
+```
+
+## Generate RSA key for a certificate
```

Comment:
what about suggesting examples with EC keys instead of old RSA keys? if show some examples, let's show new and modern instead legacy and old.

Reply:
Done